cmake/sysbuild/suit.cmake

```diff
@@ -534,12 +534,18 @@ function(suit_setup_merge)
       list(APPEND ARTIFACTS_TO_MERGE ${BINARY_DIR}/zephyr/suit_mpi_${IMAGE_TARGET_NAME}_merged.hex)
     endif()
 
+    # Get a list of files (and their dependencies) which need merging into the uicr merged file and
+    # add them at the end of the list, allowing for overwriting
+    get_property(merge_files GLOBAL PROPERTY SUIT_MERGE_FILE)
+    get_property(merge_dependencies GLOBAL PROPERTY SUIT_MERGE_DEPENDENCIES)
+
     set_property(
       GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
       COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/mergehex.py
       --overlap replace
       -o ${OUTPUT_HEX_FILE}
-       ${ARTIFACTS_TO_MERGE}
+       ${ARTIFACTS_TO_MERGE} ${merge_files}
```

Comment:
Is it intentional to merge those files multiple times (see a for loop in line 505)?

Reply:
Why is this ran multiple times? Is this for multiple cores?

Reply:
Correct, once for each domain (app and radio in case of nRF54H). This is the most challenging part of the whole mergehex process - it is important which debugger passes which HEX files, thus two uicr_merged.hex files are generated.

Reply:
So how do you know what image this is?

Reply:
We could define the `SUIT_MERGE_FILE` property independently for each image and make the caller of `suit_add_merge_hex_file` (e.g. Fast Pair CMake) responsible for indicating the target image.

However, since the Matter PR has already been merged, I suggest making this improvement in the follow-up PR.

Reply:
Since SUIT is supported on nRF54H20 only, it is assumed in this preliminary implementation that the device is the nRF54H SoC.
When it comes to the domains - any sysbuild image that has `IMAGE_TARGET_NAME` and `CONFIG_NRF_REGTOOL_GENERATE_UICR` defined is assumed to be the main image for the domain.

Reply:
Fixed

Reply:
Please note that we may have customer use cases in which the provisioned data needs to be placed in the Radio Core image together with their "main application". Some customers won't use the Application Core to minimize the latency in their radio-based applications (IPC delay). That's why I suggested adding the image argument to the `suit_add_merge_hex_file` function to make the overall provisioning solution more flexible.

However, for now, I think it is enough to support only provisioning for the main image of the Application Core.